app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt

```diff
@@ -174,16 +233,25 @@ class EndToEndTest {
         .onNodeWithTag("DescriptionField")
         .performTextInput("Teach me how to write tests pls")
     composeTestRule.onNodeWithTag("DateButton").performClick()
+    Thread.sleep(2000)
     onView(withText("OK")).perform(click())
     composeTestRule.onNodeWithTag("TimeButton").performClick()
+    Thread.sleep(2000)
     onView(withText("OK")).perform(click())
     composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
     composeTestRule.onNodeWithTag("subjectButton").performClick()
     composeTestRule.onNodeWithTag("dropdownAICC").performClick()
     composeTestRule.onNodeWithTag("mapButton").performClick()
     composeTestRule.onNodeWithTag("mapContainer").performClick()
-    Thread.sleep(2000) // Wait for the map to load
-    device.click(device.displayWidth / 2, device.displayHeight / 2)
+
+    composeTestRule.waitUntil(15000) {
+      // wait max 4 seconds for the map to load,
```

Comment:
The comment seems not completely true: if I am not mistaken it wait for max 15 seconds (not 4). Like for the commented Thread.sleep, this is more a readability issue.